### PR TITLE
Refactors the GeoLocationButton

### PR DESCRIPTION
**Pull request by KaiVolland**
### Description:

This refactors the `GeoLocationButton` to use an OlFeature instead of an OlOverlay.

### CODE REVIEW

File: src/Button/GeoLocationButton/GeoLocationButton.jsx [hunk 1]
```diff
@@ -31,6 +36,40 @@ class GeoLocationButton extends React.Component {
    */
   className = `${CSS_PREFIX}geolocationbutton`;
 
+  /**
+   * The Feature marking the current location.
```
Comment (dnlkoch) [inline src/Button/GeoLocationButton/GeoLocationButton.jsx]: `Feature` -> `feature`.

File: src/Button/GeoLocationButton/GeoLocationButton.jsx [hunk 2]
```diff
@@ -31,6 +36,40 @@ class GeoLocationButton extends React.Component {
...
+
+  /**
+   * The OpenLayers geolocation interaction.
+   */
+  _geoLocationInteraction = undefined;
+
+  /**
+   * The layer containing the markerFeature.
+   */
+  _geoLocationLayer =  new OlLayerVector({
```
Comment (dnlkoch) [inline src/Button/GeoLocationButton/GeoLocationButton.jsx]: Extra whitespace before the constructor.

File: src/Button/GeoLocationButton/GeoLocationButton.jsx [hunk 3]
```diff
@@ -99,7 +138,7 @@ class GeoLocationButton extends React.Component {
     onGeolocationChange: () => undefined,
     onError: () => undefined,
     showMarker: true,
-    follow: true,
+    follow: false,
```
Comment (dnlkoch) [inline src/Button/GeoLocationButton/GeoLocationButton.jsx]: In my opinion `follow: true` is a good default, but you decide.

File: src/Button/GeoLocationButton/GeoLocationButton.jsx [hunk 4]
```diff
@@ -114,18 +153,45 @@ class GeoLocationButton extends React.Component {
...
   constructor(props) {
     super(props);
+    const {
+      map,
+      showMarker
+    } = this.props;
 
     this.positions = new OlGeomLineString([], 'XYZM');
+    this._geoLocationLayer.setStyle(this._styleFunction);
+    map.addLayer(this._geoLocationLayer);
```
Comment (dnlkoch) [inline src/Button/GeoLocationButton/GeoLocationButton.jsx]: Please check if the layer is already added to the map. In addition I would recommend to delete the layer on unmount.

File: src/Button/GeoLocationButton/GeoLocationButton.jsx [hunk 5]
```diff
@@ -114,18 +153,45 @@ class GeoLocationButton extends React.Component {
...
+    this.state = {};
+
+    if (showMarker) {
+      this._markerFeature = new OlFeature();
+      this._geoLocationLayer.getSource().addFeature(this._markerFeature);
+    }
+  }
+
+  /**
+   * Adds the markerFeature if not allready done and adds it to the geoLocation
```
Comment (dnlkoch) [inline src/Button/GeoLocationButton/GeoLocationButton.jsx]: `allready` -> `already`.

File: src/Button/GeoLocationButton/GeoLocationButton.jsx (hunk 3 above)
Comment (KaiVolland) [reply src/Button/GeoLocationButton/GeoLocationButton.jsx]: Following is somewhat buggy atm. I think this needs a further refactoring. I'll stick with fals in the meanwhile.

Comment (dnlkoch) [reply src/Button/GeoLocationButton/GeoLocationButton.jsx]: Could you please create an issue regarding to this?

Comment (KaiVolland) [reply src/Button/GeoLocationButton/GeoLocationButton.jsx]: https://github.com/terrestris/react-geo/issues/755

**Pull request merged**